modules/core/CompositeGeometry.cpp

```diff
@@ -20,7 +19,9 @@ namespace rt {
 		, m_pBSPTree2(new CBSPTree())
 #endif
     {
-        computeBoundingBox();
+        if (operationType == BoolOp::Difference)
+			for (auto pPrim: m_vPrims2) pPrim->flipNormal();
```

Comment:
[minor]

maybe we could use `const auto&` to avoid copying the loop variable

Reply:
Yes for reference... I actually thought that there must be pointers in the vector... Let me check.

Reply:
Yes, there are pointers! I will add letter 'p' in the names of the corresponding variables.

Reply:
About adding `const`, I think it wont work, because `flipNormal()` is not const.
About reference, usually it is not needed, because a reference and a pointer usually take the same amount of memory. Here we deal with a _smart pointer_ which may be larger (I am not sure 100%). Let it be a reference.

Reply:
thanks! let's keep it as reference then :)